### PR TITLE
Fix pasting in CodeArea under mac

### LINKED ISSUE #6878: Paste in Bibtex Source field does not work
JabRef version 5.1 on OSX

- [ x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. Use the + button to add new article
2. attempt to paste a bibtex item
3. observe that paste fails

### PR DESCRIPTION
**Pull request by Siedlerchr**
CodeArea does not implement TextInputControl

Fixes #6878

It would be if someone could test this under linux and windows
1. Copy bibtex entry
2. paste in source code editor in entry editor
3. Pasting works and no duplicates on the maintable

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
The JabRef maintainers will add the following name to the [AUTHORS file](https://github.com/JabRef/jabref/blob/master/CONTRIBUTING.md#author-credits). In case you want to use a different one, please comment here and [adjust your name in your git configuration](https://docs.github.com/en/github/using-git/setting-your-username-in-git) for future commits.

```
MLEP
```

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 1]
```diff
@@ -276,7 +277,10 @@ private void initKeyBindings() {
                         break;
                     case PASTE:
                         if (OS.OS_X) { // Workaround for a jdk issue that executes paste twice when using cmd+v in a TextField
-                            event.consume();
+                            // Extra workaround for CodeArea, which does not inherit from TextInputControl
+                            if (!(stateManager.getFocusOwner().isPresent() && (stateManager.getFocusOwner().get() instanceof CodeArea))) {
+                                event.consume();
+                            }
                             break;
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/JabRefFrame.java:284]: Did not see this earlier: Should the break statement be part of the if-mac-clause?

Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/JabRefFrame.java:284]: It's part of the outer mac clause.

**Comment by calixtus** [comment]
Seems to work in Linux and Windows.

**Pull request merged**